torch_geometric/nn/conv/cugraph/gat_conv.py

```diff
@@ -61,10 +61,12 @@ def reset_parameters(self):
     def forward(
         self,
         x: Tensor,
-        csc: Tuple[Tensor, Tensor],
+        csc: Tuple[Tensor, Tensor, int],
         max_num_neighbors: Optional[int] = None,
     ) -> Tensor:
-        graph = self.get_cugraph(x.size(0), csc, max_num_neighbors)
+        assert x.size(0) == csc[
+            -1], "Size mismatch between node feature tensor and CuGraph graph."
```

Comment:
I believe this check is already done in the bindings afterwards. If needed, it would be better to improve the error messages there, since this leads to duplicated code (both the fact that the checks will be done twice and that you have to repeat this in every forward function of the different convolution modules, or create another abstraction that does this).